src/Symfony/Component/Serializer/Normalizer/DateTimeNormalizer.php

```diff
@@ -91,8 +91,8 @@ public function denormalize($data, string $type, string $format = null, array $c
         $dateTimeFormat = $context[self::FORMAT_KEY] ?? null;
         $timezone = $this->getTimezone($context);
 
-        if (null === $data || (\is_string($data) && '' === trim($data))) {
-            throw NotNormalizableValueException::createForUnexpectedDataType('The data is either an empty string or null, you should pass a string that can be parsed with the passed format or a valid DateTime string.', $data, [Type::BUILTIN_TYPE_STRING], $context['deserialization_path'] ?? null, true);
+        if (null === $data || !\is_string($data) || '' === trim($data)) {
```

Comment:
I have additionally adapted this if construct, since $data must always be a string.